### PR TITLE
refactor: locale strings desctructure

### PR DESCRIPTION
**Pull request by seekshiva**
## Type of Change

- [x] Enhancement
- [x] Refactoring

## Description

Lazy load localeStrings instead of adding it as part of main bundle.

## Motivation and Context

As we add support for more locales, it bloats up our bundle size. We can bundle a default locale (en) as part of main bundle, and lazy load everything else.

## How did you test it?

-

## Checklist

- [x] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/LocaleStrings/LocaleStringHelper.res [hunk 1]
```diff
@@ -0,0 +1,22 @@
+open LocaleStringTypes
+let mapLocalStringToTypeLocale = val => {
+  switch val {
+  | "en" => EN
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/LocaleStringHelper.res:4]:
```suggestion
```

File: src/LocaleStrings/LocaleStringHelper.res [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  | "de" => DE
+  | "fr-BE" => FR_BE
+  | "es" => ES
+  | "ca" => CA
+  | "pt" => PT
+  | "it" => IT
+  | "pl" => PL
+  | "nl" => NL
+  | "sv" => SV
+  | "ru" => RU
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/LocaleStringHelper.res:19]:
```suggestion
  | "ru" => RU
  | "en"
```

File: src/LocaleStrings/SwedishLocale.res [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    <strong> {React.string(`${currency} ${str}`)} </strong>
+    {React.string({`${Utils.nbsp}kommer att tillämpas för denna transaktion`})}
+  </>,
+  surchargeMsgAmountForCard: (currency, str) => <>
+    {React.string(`Ett tilläggsbelopp på upp till${Utils.nbsp}`)}
+    <strong> {React.string(`${currency} ${str}`)} </strong>
+    {React.string(`${Utils.nbsp}kommer att tillämpas för denna transaktion`)}
+  </>,
+  surchargeMsgAmountForOneClickWallets: "Tilläggsavgift tillkommer",
+  on: "på",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/SwedishLocale.res:71]:
```suggestion
  on: `på`,
```

File: src/LocaleStrings/ArabicLocale.res [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  billingNameLabel: `اسم الفواتير`,
+  billingNamePlaceholder: `الاسم الأول والاسم الأخير`,
+  cardHolderName: `إسم صاحب البطاقة`,
+  on: `على`,
+  \"and": `و`,
+  nameEmptyText: str => `يرجى تقديم الخاص بك ${str}`,
+  completeNameEmptyText: str => `يرجى تقديم كامل الخاص بك ${str}`,
+  billingDetailsText: `تفاصيل الفاتورة`,
+  socialSecurityNumberLabel: `رقم الضمان الاجتماعي`,
+  saveWalletDetails: "سيتم حفظ تفاصيل المحفظة عند الاختيار",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/ArabicLocale.res:77]:
```suggestion
  saveWalletDetails: `سيتم حفظ تفاصيل المحفظة عند الاختيار`,
```

File: src/LocaleStrings/DeutschLocale.res [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  billingNameLabel: `Abrechnungsname`,
+  billingNamePlaceholder: `Vor-und Nachname`,
+  cardHolderName: `Name des Karteninhabers`,
+  on: `An`,
+  \"and": `Und`,
+  nameEmptyText: str => `Bitte geben Sie Ihre an ${str}`,
+  completeNameEmptyText: str => `Bitte geben Sie Ihr vollständiges Formular an ${str}`,
+  billingDetailsText: `Rechnungsdetails`,
+  socialSecurityNumberLabel: `Sozialversicherungsnummer`,
+  saveWalletDetails: "Wallet-Details werden beim Auswählen gespeichert",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/DeutschLocale.res:77]:
```suggestion
  saveWalletDetails: `Wallet-Details werden beim Auswählen gespeichert`,
```

File: src/LocaleStrings/FrenchLocale.res [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  billingNameLabel: `Nom de facturation`,
+  billingNamePlaceholder: `Prénom et nom de famille`,
+  cardHolderName: `Nom du titulaire`,
+  on: `sur`,
+  \"and": `et`,
+  nameEmptyText: str => `Veuillez fournir votre ${str}`,
+  completeNameEmptyText: str => `Veuillez fournir votre complet ${str}`,
+  billingDetailsText: `Détails de la facturation`,
+  socialSecurityNumberLabel: `Numéro de sécurité sociale`,
+  saveWalletDetails: "Les détails du portefeuille seront enregistrés lors de la sélection",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/FrenchLocale.res:77]:
```suggestion
  saveWalletDetails: `Les détails du portefeuille seront enregistrés lors de la sélection`,
```

File: src/LocaleStrings/HebrewLocale.res [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+  billingNameLabel: `שם החיוב`,
+  billingNamePlaceholder: `שם פרטי ושם משפחה`,
+  cardHolderName: `שם בעל הכרטיס`,
+  on: `עַל`,
+  \"and": `ו`,
+  nameEmptyText: str => `אנא ספק את שלך ${str}`,
+  completeNameEmptyText: str => `אנא ספק את המלא שלך ${str}`,
+  billingDetailsText: `פרטי תשלום`,
+  socialSecurityNumberLabel: `מספר ביטוח לאומי`,
+  saveWalletDetails: "פרטי הארנק יישמרו בעת בחירה",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/HebrewLocale.res:77]:
```suggestion
  saveWalletDetails: `פרטי הארנק יישמרו בעת בחירה`,
```

File: src/LocaleStrings/JapaneseLocale.res [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+  postalCodeEmptyText: `郵便番号を空白にすることはできません`,
+  postalCodeInvalidText: `郵便番号が無効です`,
+  stateLabel: `州`,
+  stateEmptyText: `状態を空にすることはできません`,
+  cityLabel: `街`,
+  line2Placeholder: `アパート、ユニット番号など（任意）`,
+  line2EmptyText: `住所行 2 を空にすることはできません`,
+  countryLabel: `国`,
+  cityEmptyText: `都市を空にすることはできません`,
+  currencyLabel: "通貨",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/JapaneseLocale.res:33]:
```suggestion
  currencyLabel: `通貨`,
```

File: src/LocaleStrings/JapaneseLocale.res [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+  billingNameLabel: `課金名`,
+  billingNamePlaceholder: `名前と苗字`,
+  cardHolderName: `クレジットカード名義人氏名`,
+  on: `の上`,
+  \"and": `そして`,
+  nameEmptyText: str => `あなたの情報を提供してください ${str}`,
+  completeNameEmptyText: str => `完全な情報を提供してください ${str}`,
+  billingDetailsText: `支払明細`,
+  socialSecurityNumberLabel: `社会保障番号`,
+  saveWalletDetails: "選択時にウォレットの詳細が保存されます",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/JapaneseLocale.res:77]:
```suggestion
  saveWalletDetails: `選択時にウォレットの詳細が保存されます`,
```

File: src/LocaleStrings/SwedishLocale.res [hunk 10]
```diff
@@ -0,0 +1,82 @@
...
+  cardNickname: `Kortets smeknamn`,
+  billingNamePlaceholder: `Förnamn och efternamn`,
+  emailEmptyText: `E-post får inte vara tom`,
+  emailInvalidText: `Ogiltig e-postadress`,
+  line1EmptyText: `Adressrad 1 får inte vara tom`,
+  line2EmptyText: `Adressrad 2 får inte vara tom`,
+  cityEmptyText: `Staden får inte vara tom`,
+  postalCodeEmptyText: `Postnummer får inte vara tomt`,
+  postalCodeInvalidText: `Ogiltigt postnummer`,
+  stateEmptyText: "Staten får inte vara tom",
```
Comment (vsrivatsa-juspay) [inline src/LocaleStrings/SwedishLocale.res:59]:
```suggestion
  stateEmptyText: `Staten får inte vara tom`,
```

**Review by vsrivatsa-juspay — CHANGES_REQUESTED** [review]
Please address comments

**Review by prafulkoppalkar — DISMISSED** [review]

File: src/LocaleStrings/LocaleStringHelper.res (hunk 2 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/LocaleStringHelper.res:19]: Resolved

File: src/LocaleStrings/SwedishLocale.res (hunk 3 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/SwedishLocale.res:71]: Resolved

File: src/LocaleStrings/DeutschLocale.res (hunk 5 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/DeutschLocale.res:77]: Resolved

File: src/LocaleStrings/FrenchLocale.res (hunk 6 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/FrenchLocale.res:77]: Resolved

File: src/LocaleStrings/HebrewLocale.res (hunk 7 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/HebrewLocale.res:77]: Resolved

File: src/LocaleStrings/JapaneseLocale.res (hunk 8 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/JapaneseLocale.res:33]: Resolved

File: src/LocaleStrings/ArabicLocale.res (hunk 4 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/ArabicLocale.res:77]: Resolved

File: src/LocaleStrings/JapaneseLocale.res (hunk 9 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/JapaneseLocale.res:77]: Resolved

File: src/LocaleStrings/SwedishLocale.res (hunk 10 above)
Comment (PritishBudhiraja) [reply src/LocaleStrings/SwedishLocale.res:59]: Resolved

**Review by PritishBudhiraja — APPROVED** [review]

**Review by prafulkoppalkar — APPROVED** [review]

**Review by vsrivatsa-juspay — APPROVED** [review]

**Pull request merged**